common/src/main/java/org/conscrypt/HpkeSuite.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.conscrypt;
+
+import java.security.PrivateKey;
+import java.security.PublicKey;
+
+/**
+ * Holds the KEM, KDF, and AEAD that are used and supported by {@link Hpke} defined on
+ * RFC 9180.
+ *
+ * <ul>
+ *   <li><a
+ * href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">KEM</a></li>
+ *   <li><a
+ * href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-derivation-functions-kd">KDF</a></li>
+ *   <li><a
+ * href="https://www.rfc-editor.org/rfc/rfc9180.html#name-authenticated-encryption-wi">AEAD</a></li>
+ * </ul>
+ */
+public final class HpkeSuite {
+  /**
+   * {@link HpkeSuite} with the following algorithm scheme:
+   * <li>
+   *      <ul>KEM:  0x0020: DHKEM(X25519, HKDF-SHA256)</ul>
+   *      <ul>KDF:  0x0001: HKDF-SHA256</ul>
+   *      <ul>AEAD: 0x0001: AES-128-GCM</ul>
+   * </li>
+   *
+   * @see <a
+   * href="https://www.rfc-editor.org/rfc/rfc9180
+   * .html#name-key-encapsulation-mechanism">KEMs</a>
+   * @see <a
+   * href="https://www.rfc-editor.org/rfc/rfc9180
+   * .html#name-key-derivation-functions-kd">KDFs</a>
+   * @see <a
+   * href="https://www.rfc-editor.org/rfc/rfc9180
+   * .html#name-authenticated-encryption-wi">AEAD</a>
+   */
+  public static final HpkeSuite DHKEM_X25519_HKDF_SHA256_HKDF_SHA256_AES_128_GCM =
+      new HpkeSuite(KEM.DHKEM_X25519_HKDF_SHA256, KDF.HKDF_SHA256, AEAD.AES_128_GCM);
```

Comment:
Is the expectation in this design that Conscrypt needs to be updated every time someone comes up with a new combined HPKE suite? That doesn't match the specification and seems it would cause combinatorial extra work when, say, a PQ KEM gets added.

(Were HPKE a scheme where suites were pre-composed, that would have been reasonable. I certainly wouldn't have minded there being fewer combinations! But that's not what we have and we shouldn't diverge fromt he spec without reason.)

Reply:
I have updated HpkeSuite constructor to accept KEM, KDF, and AEAD, like:
```java
new HpkeSuite(KEM_DHKEM_X25519_HKDF_SHA256, KDF_HKDF_SHA256, AEAD_AES_256_GCM);
```

Each parameter is an int so it could be expanded later one. WDYT?

Reply:
SGTM. For C/C++ we prefer objects over ints because it helps with dead code elimination, but that's already a lost cause here. :-)